cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -1,56 +1,40 @@
-import logging
 from http import HTTPStatus
+from pathlib import Path
 
-from pydantic import ValidationError
-from requests import RequestException, Response, Session
+from requests import Session
 from requests.adapters import HTTPAdapter
 from urllib3 import Retry
 
 from cg.clients.freshdesk.constants import EndPoints
-from cg.clients.freshdesk.exceptions import (
-    FreshdeskAPIException,
-    FreshdeskModelException,
-)
-from cg.clients.freshdesk.models import TicketCreate, TicketResponse
-
-LOG = logging.getLogger(__name__)
+from cg.clients.freshdesk.models import ReplyCreate, TicketCreate, TicketResponse
+from cg.clients.freshdesk.utils import handle_client_errors, prepare_attachments
 
 
 class FreshdeskClient:
     """Client for communicating with the freshdesk REST API."""
 
-    def __init__(self, base_url: str, api_key: str):
-        self.base_url = base_url
-        self.api_key = api_key
+    def __init__(self, config: dict):
+        self.base_url = config["freshdesk_url"]
+        self.api_key = config["freshdesk_api_key"]
```

Comment:
Passing around dicts on constructors hides what parameters are needed for the class to be instantiated and creates "magic strings" for fetching the correct keys. I think we should explicitly pass the url and the key

Reply:
Thanks for the clarification! ✅